net/kcptun/files/kcptun-c.init

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2019 Dengfeng Liu
+
+START=99
+
+USE_PROCD=1
+NAME=kcptun-c
+PROG=/usr/bin/${NAME}
+
+validate_section_kcptun() 
+{
+	uci_load_validate "${NAME}" kcptun "$1" "$2" \
+		'local_port:port' \
+		'remote_ip:string' \
+		'remote_port:port' \
+		'mode:string' \
+		'nocomp:bool' \
+		'sndwnd:uinteger' \
+		'rcvwnd:uinteger' \
+		'disabled:bool'
+}
+
+kcptun_instance() 
+{
+	[ "$2" = 0 ] || {
+		echo "validation failed"
+		return 1
+	}
+
+	[ "${disabled}" = "1" ] && return 1
+
+	[ "${local_port}" -gt 0 -a "${local_port}" -lt 65535 ] || {
```

Comment:
These two throw this warning under shellcheck: https://github.com/koalaman/shellcheck/wiki/SC2166

I also don't like this. Ports below 1024 are usually reserved. I saw this in a different project:

if ! echo ${PORT} | grep -E "^[1-9][0-9]{1,4}$"; then

Could be useful here.

Reply:
Sorry, just noticed this... `^[1-9][0-9]{1,4}$` would test if the value is 10-99999, yes? This doesn't seem safe :joy:

While ports < 1024 are usually reserved, I think there are cases where it can be justified, i.e. binding to port 80 or 443 to bypass (tunnel through) naive firewalls. (And since the current init script runs the client/server as root, it can bind to those ports.)

@neheb Would you be okay with this:

```sh
[ "$local_port" -gt 0 ] && [ "$local_port" -lt 65536 ] || return 1
```

Reply:
That throws a different error. C can run while A is true but not B.

Reply:
`return 1` should run if `[ "$local_port" -lt 65536 ]` is false, because then `$local_port` is not a valid port number. (This isn't a replacement for `A ? B : C`, this is `(A && B) || C`.)

Reply:
Just checked. It's not throwing a warning.

Reply:
I was referring to this earlier: https://github.com/koalaman/shellcheck/wiki/SC2015

Reply:
I have tested
`[ "$local_port" -gt 0 ] && [ "$local_port" -lt 65536 ] || return 1`
will be ok, and shellcheck without warning